openstack/workflow/v2/executions/requests.go

```diff
@@ -64,3 +73,165 @@ func Delete(client *gophercloud.ServiceClient, id string) (r DeleteResult) {
+type ListOptsBuilder interface {
+	ToExecutionListQuery() (string, error)
+}
+
+// ListOpts filters the result returned by the List() function.
+type ListOpts struct {
+	// WorkflowName allows to filter by workflow name.
+	WorkflowName *ListFilter `q:"-"`
+	// WorkflowID allows to filter by workflow id.
+	WorkflowID string `q:"workflow_id"`
+	// Description allows to filter by execution description.
+	Description *ListFilter `q:"-"`
+	// Params allows to filter by specific parameters.
+	Params map[string]interface{} `q:"-"`
+	// TaskExecutionID allows to filter with a specific task execution id.
+	TaskExecutionID string `q:"task_execution_id"`
+	// RootExecutionID allows to filter with a specific root execution id.
+	RootExecutionID string `q:"root_execution_id"`
+	// State allows to filter by execution state.
+	// Possible values are IDLE, RUNNING, PAUSED, SUCCESS, ERROR, CANCELLED.
+	State *ListFilter `q:"-"`
+	// StateInfo allows to filter by state info.
+	StateInfo *ListFilter `q:"-"`
+	// Input allows to filter by specific input.
+	Input map[string]interface{} `q:"-"`
+	// Output allows to filter by specific output.
+	Output map[string]interface{} `q:"-"`
+	// CreatedAt allows to filter by execution creation date.
+	CreatedAt *ListDateFilter `q:"-"`
+	// UpdatedAt allows to filter by last execution update date.
+	UpdatedAt *ListDateFilter `q:"-"`
+	// IncludeOutput requests to include the output for all executions in the list.
+	IncludeOutput bool `q:"-"`
+	// ProjectID allows to filter by given project id. Admin required.
+	ProjectID string `q:"project_id"`
+	// AllProjects requests to get executions of all projects. Admin required.
+	AllProjects int `q:"all_projects"`
+	// SortDir allows to select sort direction.
+	// It can be "asc" or "desc" (default).
+	SortDirs string `q:"sort_dirs"`
+	// SortKey allows to sort by one of the execution attributes.
+	SortKeys string `q:"sort_keys"`
+	// Marker and Limit control paging.
+	// Marker instructs List where to start listing from.
+	Marker string `q:"marker"`
+	// Limit instructs List to refrain from sending excessively large lists of
+	// executions.
+	Limit int `q:"limit"`
+}
+
+// ListFilter allows to filter string parameters with different filters.
+// Empty value for Filter checks for equality.
+type ListFilter struct {
+	Filter FilterType
+	Value  string
+}
+
+func (l ListFilter) String() string {
+	return fmt.Sprintf("%s%s", l.Filter, l.Value)
+}
+
+// ListDateFilter allows to filter date parameters with different filters.
+// Empty value for Filter checks for equality.
+type ListDateFilter struct {
+	Filter FilterType
+	Value  time.Time
+}
+
+func (l ListDateFilter) String() string {
+	return fmt.Sprintf("%s%s", l.Filter, l.Value.Format(gophercloud.RFC3339ZNoTNoZ))
+}
+
+// FilterType represents a valid filter to use for filtering executions.
+type FilterType string
+
+const (
+	// FilterEq checks equality.
+	FilterEq = "eq"
```

Comment:
Let's use capital letters where applicable, like how it's done here: https://github.com/gophercloud/gophercloud/blob/master/openstack/imageservice/v2/images/types.go#L90-L95

So, capitalize everything except `Has` since that's an actual word. Change `NotEq` to `NEQ`, etc.